packages/core/src/utils/debugLogger.ts

```diff
@@ -0,0 +1,66 @@
+const CF_DEBUG_KEY = 'cf_debug';
+
+export class DebugLogger {
+  private static instance: DebugLogger | null = null;
+  private enabled: boolean;
+
+  constructor() {
+    if (typeof localStorage === 'undefined') {
+      this.enabled = false;
+      return;
+    }
+
+    // Default to checking localStorage for the debug mode on initialization if in browser
+    this.enabled = localStorage.getItem(CF_DEBUG_KEY) === 'true';
+  }
+
+  public static getInstance(): DebugLogger {
+    if (this.instance === null) {
+      this.instance = new DebugLogger();
+    }
+    return this.instance;
+  }
+
+  public getEnabled() {
+    return this.enabled;
+  }
+
+  public setEnabled(enabled: boolean) {
+    this.enabled = enabled;
+
+    if (typeof localStorage === 'undefined') {
+      return;
+    }
+    if (enabled) {
+      localStorage.setItem(CF_DEBUG_KEY, 'true');
+    } else {
+      localStorage.removeItem(CF_DEBUG_KEY);
+    }
+  }
+
+  // Log method for different levels (error, warn, log)
+  private logger(level: 'error' | 'log' | 'warn'): typeof console.log {
```

Comment:
For the native console, there are [4 levels](https://developer.chrome.com/docs/devtools/console/reference#level): verbose > info > warning > error.

I think it would be great to have this kind of separation as well. So we could separate between one-time-info messages (e.g. "Tree updated") and the spamming verbose message that would fire for every node or every property.